packages/cfd/src/Components/__tests__/cfd-demo-account-display.spec.js

```diff
@@ -0,0 +1,361 @@
+import { fireEvent, render, screen, within } from '@testing-library/react';
+import React from 'react';
+import { CFDDemoAccountDisplay } from '../cfd-demo-account-display';
+
+describe('<CFDDemoAccountDisplay />', () => {
+    const TESTED_CASES = {
+        EU: 'eu',
+        LOADING: 'loading',
+        NON_EU_DMT5: 'non_eu_dmt5',
+        NON_EU_DXTRADE: 'non_eu_dxtrade',
+    };
+
+    let props;
+
+    beforeEach(() => {
+        props = {
+            current_list: {},
```

Comment:
since the type is array, shouldn't this be an empty array instead of object?

Reply:
current_list is not an array. It's a keyed collection. It contains keys and values because this is how someone chose to fill this object/array with data. Also, It always have 0 length because we never fill it by its indexes or with a push() method.
Also, my tests would not pass if it was an array.

Reply:
checked, you're right